data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorUrlServiceIT.java

```diff
@@ -0,0 +1,103 @@
+
+        maxmindLicenseKey = System.getProperty("tests.geoipProcessor.maxmindLicenseKey");
+
+        jsonInput = "{\"peer\": {\"ip\": \"8.8.8.8\", \"host\": \"example.org\" }, \"status\": \"success\"}";
+
+        geoIPInputJson = objectMapper.readValue(jsonInput, GeoIPInputJson.class);
+        tempPath = System.getProperty("java.io.tmpdir")+ File.separator + TEMP_PATH_FOLDER;
+
+        String asnUrl = "https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-ASN&license_key=" + maxmindLicenseKey + URL_SUFFIX;
+        String cityUrl = "https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key=" + maxmindLicenseKey + URL_SUFFIX;
+        String countryUrl = "https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-Country&license_key=" + maxmindLicenseKey + URL_SUFFIX;
+
+        String pipelineConfig = "        aws:\n" +
+                "          region: us-east-2\n" +
+                "          sts_role_arn: \"arn:aws:iam::123456789:role/data-prepper-execution-role\"\n" +
+                "        keys:\n" +
+                "          - key:\n" +
+                "              source: \"/peer/ip\"\n" +
+                "              target: \"target1\"\n" +
+                "          - key:\n" +
+                "              source: \"/peer/ip2\"\n" +
+                "              target: \"target2\"\n" +
+                "              attributes: [\"city_name\",\"country_name\"]\n" +
+                "        service_type:\n" +
+                "          maxmind:\n" +
+                "            database_path:\n" +
+                "              - url: " + asnUrl + "\n" +
+                "              - url: " + cityUrl + "\n" +
+                "              - url: " + countryUrl + "\n" +
+                "            load_type: \"cache\"\n" +
+                "            cache_size: 8192\n" +
+                "            cache_refresh_schedule: PT3M";
+
+        objectMapper.registerModule(new JavaTimeModule());
+        this.geoIPProcessorConfig = objectMapper.readValue(pipelineConfig, GeoIPProcessorConfig.class);
+    }
+
+    public GeoIPProcessorService createObjectUnderTest() {
+        return new GeoIPProcessorService(geoIPProcessorConfig, tempPath);
+    }
+
+    @Test
+    void verify_enrichment_of_data_from_maxmind_url() throws UnknownHostException {
+
+        Map<String, Object> geoData = new HashMap<>();
+        this.geoIPProcessorService = createObjectUnderTest();
+        String ipAddress = geoIPInputJson.getPeer().getIp();
+        if (IPValidationcheck.isPublicIpAddress(ipAddress)) {
+            InetAddress inetAddress = InetAddress.getByName(ipAddress);
+            //All attributes are considered by default with null
+            geoData = geoIPProcessorService.getGeoData(inetAddress, null);
+
+            assertThat(geoData.get("country_iso_code"), equalTo("US"));
+            assertThat(geoData.get("ip"), equalTo("/8.8.8.8"));
```

Comment:
Why is it "/8.8.8.8" and not "8.8.8.8"?

Reply:
Updated the code to get the valid IP in PR#2925 also updated assertion statement with "8.8.8.8" instead of "/8.8.8.8".